crates/astria-merkle/tests/inclusion.rs

```diff
@@ -0,0 +1,100 @@
+//! Tests taken and adjusted to Rust from [transparency-dev/merkle].
+//!
+//! [transparency-dev/merkle]: https://github.com/transparency-dev/merkle/blob/7d6ba6631786ace38b5ccf5b0cbb31bff5c70e25/testonly/reference_test.go
+
+use astria_merkle::Tree;
+use hex_literal::hex;
+
+const LEAF_INPUTS: &[&[u8]] = &[
+    &hex!(""),
+    &hex!("00"),
+    &hex!("10"),
+    &hex!("2021"),
+    &hex!("3031"),
+    &hex!("40414243"),
+    &hex!("5051525354555657"),
+    &hex!("606162636465666768696a6b6c6d6e6f"),
+];
+
+struct InclusionTest {
+    num_leaves: usize,
+    leaf: usize,
+    expected_proof: &'static [u8],
+}
+
+const INCLUSION_TESTS: &[InclusionTest] = &[
```

Comment:
can you add a test for the empty tree (leaves = 0) also?

Reply:
I added an example to the `Tree::construct_proof` doctest that demonstrates the empty tree leading to `None`.